pkg/controller/config/config_controller.go

```diff
@@ -292,42 +294,72 @@ func (r *ReconcileConfig) validatePipeline(req reconcile.Request, cfg *op.Config
 	log := requestLogger(req, "validate-pipeline")
 	log.Info("validating pipelines")
 
+	running, err := r.validateDeployments(req, cfg)
+	if err != nil {
+		log.Error(err, "failed to validate pipeline controller deployments")
+		_ = r.updateStatus(cfg, op.ConfigCondition{
+			Code:    op.PipelineValidateError,
+			Details: err.Error(),
+			Version: flag.TektonVersion})
+		return reconcile.Result{}, err
+	}
+
+	if !running {
+		return reconcile.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
+	}
+
+	found, err := validate.Webhook(context.TODO(), r.client, flag.PipelineWebhookConfiguration)
+	if err != nil {
+		log.Error(err, "failed to validate mutating webhook")
+		_ = r.updateStatus(cfg, op.ConfigCondition{
+			Code:    op.PipelineValidateError,
+			Details: err.Error(),
+			Version: flag.TektonVersion})
+		return reconcile.Result{RequeueAfter: 15 * time.Second}, err
+	}
+	if !found {
+		return reconcile.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
+	}
+
+	err = r.updateStatus(cfg, op.ConfigCondition{Code: op.ValidatedPipeline, Version: flag.TektonVersion})
+	if err != nil {
+		return reconcile.Result{}, err
+
+	}
+	// requeue with delay for services to be up and running
+	return reconcile.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
+}
+
+func (r *ReconcileConfig) validateDeployments(req reconcile.Request, cfg *op.Config) (bool, error) {
+	log := requestLogger(req, "validate-pipeline").WithName("deployments")
+	log.Info("validating pipelines controller")
+
 	controller, err := validate.Deployment(context.TODO(),
 		r.client,
 		flag.PipelineControllerName,
 		cfg.Spec.TargetNamespace,
 	)
 	if err != nil {
-		log.Error(err, "failed to validate pipeline deployment")
-		_ = r.updateStatus(cfg, op.ConfigCondition{
-			Code:    op.PipelineError,
-			Details: err.Error(),
-			Version: flag.TektonVersion})
-		return reconcile.Result{}, err
+		log.Error(err, "validating controller deployment error")
+		return false, err
 	}
+
+	log.Info("validating webhook")
 	webhook, err := validate.Deployment(context.TODO(),
 		r.client,
 		flag.PipelineWebhookName,
 		cfg.Spec.TargetNamespace,
 	)
 	if err != nil {
-		log.Error(err, "failed to validate pipeline deployment")
-		_ = r.updateStatus(cfg, op.ConfigCondition{
-			Code:    op.PipelineError,
-			Details: err.Error(),
-			Version: flag.TektonVersion})
-		return reconcile.Result{}, err
+		log.Error(err, "validating webhook deployment error")
+		return false, err
 	}
+
 	if !controller || !webhook {
-		log.Info("pipeline controller or/and pipeline webhook is not up")
-		return reconcile.Result{
-			Requeue:      true,
-			RequeueAfter: 5 * time.Second,
```

Comment:
@hrishin the above Reque{REqueafter: 15} is a replacement for this.
Now it is being separated for tekton-webhook and tekton-controller